src/main/java/Tasks/Tasks.java

```diff
@@ -0,0 +1,79 @@
+package Tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * A class that belongs to the Tasks Package.
+ * This class encapsulates the logic of how Tasks should be represented in Duke.
+ */
+abstract public class Tasks {
+
+    private String task;
+    private Boolean marked;
+
+    /**
+     * Constructor for Tasks.
+     * @param task Task that should be recorded.
+     * @param marked Boolean flag for whether a task is completed.
+     */
+    public Tasks(String task, Boolean marked) {
+        this.task = task;
+        this.marked = marked;
+    }
+
+    /**
+     * Sets {@link #marked} to the specified value.
+     * @param var Boolean to mark a task as completed/not completed.
+     */
+    public void setMarked(Boolean var) {
+        this.marked = var;
+    }
+
+    /**
+     * Gets {@link #task}
+     * @return {@link #task}.
+     */
+    public String getTask() {
+        return this.task;
+    }
+
+    /**
+     * Gets {@link #marked}.
+     * @return {@link #marked}.
+     */
+    public boolean getMarked() {
+        return this.marked;
+    }
+
+
+    /**
+     * Abstract method to specify how a String representation of Task should be cached into
+     * a file.
+     * @return
+     */
+    abstract public String cacheString();
+
+    /**
+     * Converts a date of format "yyyy-MM-dd" to "MMM dd yyyy".
+     * @param s String representation of a date.
+     * @return Date that follows the format if applicable.
+     */
+    public static String returnDate(String s) {
```

Comment:
Function name could be better named to describe the function (To convert date from one form to another). May I suggest using convertDate instead?